frontend/src/api/k8s/__tests__/configMaps.spec.ts

```diff
@@ -0,0 +1,237 @@
+import {
+  k8sCreateResource,
+  k8sDeleteResource,
+  k8sGetResource,
+  k8sUpdateResource,
+} from '@openshift/dynamic-plugin-sdk-utils';
+import { mock200Status, mock404Error } from '~/__mocks__/mockK8sStatus';
+import {
+  assembleConfigMap,
+  createConfigMap,
+  deleteConfigMap,
+  getConfigMap,
+  replaceConfigMap,
+} from '~/api/k8s/configMaps';
+import { ConfigMapModel } from '~/api/models';
+import { ConfigMapKind, KnownLabels } from '~/k8sTypes';
+
+jest.mock('@openshift/dynamic-plugin-sdk-utils', () => ({
+  k8sGetResource: jest.fn(),
+  k8sCreateResource: jest.fn(),
+  k8sUpdateResource: jest.fn(),
+  k8sDeleteResource: jest.fn(),
+}));
+
+const k8sGetResourceMock = k8sGetResource as jest.Mock;
+const k8sCreateResourceMock = k8sCreateResource as jest.Mock;
+const k8sUpdateResourceMock = k8sUpdateResource as jest.Mock;
+const k8sDeleteResourceMock = k8sDeleteResource as jest.Mock;
+
+const projectName = 'testProject';
+const configMapName = 'test';
+const configMapMock = {
+  kind: 'ConfigMap',
+  apiVersion: 'v1',
+  metadata: {
+    name: configMapName,
+    labels: { [KnownLabels.DASHBOARD_RESOURCE]: 'true' },
+    namespace: projectName,
+  },
+  data: {
+    key1: 'value1',
+    key2: 'value2',
+  },
+} as ConfigMapKind;
+
+const configMapData = {
```

Comment:
Same as the other PR ,
I can do this 
`const {data}=configMapMock.data` but here I am getting type error   
```
Argument of type '({ [key: string]: unknown; } & Record<string, string>) | undefined' is not assignable to parameter of type 'Record<string, string>'.
  Type 'undefined' is not assignable to type 'Record<string, string>
```
Should I do this in this way 
`const { data = {} } = configMapMock;`

Reply:
We should have a separate `mockConfigMap` utility. It could accept a data param. In doing so you are sourcing a mock with the same data value.